src/cli/commands/deploy.ts

```diff
@@ -120,6 +138,33 @@ export async function deploy(options: SWACLIConfig) {
       logger.log(`  ${chalk.green(resolvedApiLocation)}`);
       logger.log(``);
     }
+    if (!apiLanguage) {
+      apiLanguage = DEFAULT_CONFIG.apiLanguage;
+      apiVersion = DEFAULT_CONFIG.apiVersion;
+      logger.silly(`Consider providing api-language and version using --api-language and --api-version flags,
+      otherwise default values apiLanguage: ${apiLanguage} and apiVersion: ${apiVersion} will apply`);
+    } else if (apiLanguage && !apiVersion) {
+      switch (apiLanguage) {
+        case "node":
+          apiVersion = DEFAULT_NODE_VERSION;
+          break;
+        case "python":
+          apiVersion = DEFAULT_PYTHON_VERSION;
+          break;
+        case "dotnet":
+          apiVersion = DEFAULT_DOTNET_VERSION;
+          break;
+        case "dotnetisolated":
+          apiVersion = DEFAULT_DOTNET_ISOLATED_VERSION;
+          break;
+        default:
+          apiLanguage = DEFAULT_CONFIG.apiLanguage;
+          apiVersion = DEFAULT_CONFIG.apiVersion;
+          break;
+      }
+      logger.silly(`Api language ${apiLanguage} is provided but api version is not provided.
+      Assuming default version ${apiVersion}`);
+    }
```

Comment:
Can we move this to swa init? WDYT @sinedied?

Reply:
Wassim, which place makes more sense for doing this?

Reply:
@manekinekko definitely, I answered where this should go in the above comment